### DISCUSSION TITLE
const in CBLAS_LAYOUT et al. in CBLAS

### DISCUSSION THREAD

**Discussion by jgpallero** [Reference-LAPACK/lapack#841] [Q&A]
Inspecting the cblas.h file I can see that the qualifier `const` is employed in all input arguments (even with those called by value) with the exception of the enumerated types CBLAS_LAYOUT et al. Two questions:

1. Why `const` is not used with the enumerated types?
2. Function `cblas_sgemv` maintains `const` with the enumerated types. I suppose it is a forget from previous versions (I seem to remember that `const` was used in previous versions). Am I right?

**Comment by weslleyspereira** [comment]
> Why const is not used with the enumerated types?

I don't think there is a good reason for that. Maybe for improving readability by reducing verbosity. I am not aware of any performance gain for using `const` in passing variables by value. By not using `const` you allow for the variable to be modified inside the routine, but I don't think this was the original reason.
